<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>714</width>
    <height>561</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Filter and Derivatives</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QScrollArea" name="scrollAreaSource">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QcvMatWidget" name="sourceImage">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>688</width>
         <height>309</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidgetFilters">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabImage">
       <attribute name="title">
        <string>Image</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>8</number>
        </property>
        <property name="bottomMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBoxSize">
          <property name="title">
           <string>Size</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="leftMargin">
            <number>8</number>
           </property>
           <property name="topMargin">
            <number>8</number>
           </property>
           <property name="rightMargin">
            <number>8</number>
           </property>
           <property name="bottomMargin">
            <number>8</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QRadioButton" name="radioButtonOrigSize">
               <property name="text">
                <string>Original</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioButtonCustomSize">
               <property name="text">
                <string>Custom</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLabel" name="labelWidth">
                 <property name="text">
                  <string>Width</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBoxWidth">
                 <property name="maximum">
                  <number>1600</number>
                 </property>
                 <property name="singleStep">
                  <number>4</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLabel" name="labelHeight">
                 <property name="text">
                  <string>Height</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBoxHeight">
                 <property name="maximum">
                  <number>1200</number>
                 </property>
                 <property name="singleStep">
                  <number>4</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxConv">
          <property name="title">
           <string>Conversions</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="leftMargin">
            <number>8</number>
           </property>
           <property name="topMargin">
            <number>8</number>
           </property>
           <property name="rightMargin">
            <number>8</number>
           </property>
           <property name="bottomMargin">
            <number>8</number>
           </property>
           <item>
            <widget class="QCheckBox" name="checkBoxFlip">
             <property name="text">
              <string>Flip</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxImages">
             <item>
              <property name="text">
               <string>Input</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Gray</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Blurred</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>X Gradient</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Y Gradient</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Gradient Magnitude</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Gradient Angle</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Edge Map</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Laplacian</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Cornerness</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Harris Cornerness</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxEdges">
             <item>
              <property name="text">
               <string>Threshold</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Canny</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Merged</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabVariables">
       <attribute name="title">
        <string>Variables</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>8</number>
        </property>
        <property name="bottomMargin">
         <number>8</number>
        </property>
        <item row="1" column="0">
         <widget class="QLabel" name="labelSigma">
          <property name="text">
           <string>Sigma</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="labelSigmaMin">
          <property name="text">
           <string>0.0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSlider" name="horizontalSliderSigma">
          <property name="maximum">
           <number>315</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="labelSigmaMax">
          <property name="text">
           <string>1.0</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QDoubleSpinBox" name="doubleSpinBoxSigma">
          <property name="minimum">
           <double>0.350000000000000</double>
          </property>
          <property name="maximum">
           <double>3.500000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelKernel">
          <property name="text">
           <string>Kernel</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="labelKernelMin">
          <property name="text">
           <string>3</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSlider" name="horizontalSliderKernel">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>15</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="pageStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="labelKernelMax">
          <property name="text">
           <string>15</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QSpinBox" name="spinBoxKernel">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>15</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelThreshold">
          <property name="text">
           <string>Threshold</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="labelThresholdMin">
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QSlider" name="horizontalSliderThreshold">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>127</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QLabel" name="labelThresholdMax">
          <property name="text">
           <string>255</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QSpinBox" name="spinBoxThreshold">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>127</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="labelKappa">
          <property name="text">
           <string>Kappa</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="labelKappaMin">
          <property name="text">
           <string>0.04</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSlider" name="horizontalSliderKappa">
          <property name="maximum">
           <number>11</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QLabel" name="labelKappaMax">
          <property name="text">
           <string>0.15</string>
          </property>
         </widget>
        </item>
        <item row="3" column="4">
         <widget class="QDoubleSpinBox" name="doubleSpinBoxKappa">
          <property name="minimum">
           <double>0.040000000000000</double>
          </property>
          <property name="maximum">
           <double>0.150000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>714</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSources">
    <property name="title">
     <string>Sources</string>
    </property>
    <addaction name="actionCamera_0"/>
    <addaction name="actionCamera_1"/>
    <addaction name="actionFile"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuVideo">
    <property name="title">
     <string>Video</string>
    </property>
    <widget class="QMenu" name="menuSize">
     <property name="title">
      <string>taille</string>
     </property>
     <addaction name="actionOriginalSize"/>
     <addaction name="actionConstrainedSize"/>
    </widget>
    <addaction name="actionFlip"/>
    <addaction name="separator"/>
    <addaction name="menuSize"/>
   </widget>
   <widget class="QMenu" name="menuRender">
    <property name="title">
     <string>Render</string>
    </property>
    <addaction name="actionRenderImage"/>
    <addaction name="actionRenderPixmap"/>
    <addaction name="actionRenderOpenGL"/>
   </widget>
   <addaction name="menuSources"/>
   <addaction name="menuVideo"/>
   <addaction name="menuRender"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionCamera_0">
   <property name="text">
    <string>Camera 0</string>
   </property>
   <property name="toolTip">
    <string>caméra interne</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+0</string>
   </property>
  </action>
  <action name="actionCamera_1">
   <property name="text">
    <string>Camera 1</string>
   </property>
   <property name="toolTip">
    <string>caméra externe</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+1</string>
   </property>
  </action>
  <action name="actionFile">
   <property name="text">
    <string>File</string>
   </property>
   <property name="toolTip">
    <string>fichier video</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionFlip">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>flip</string>
   </property>
  </action>
  <action name="actionOriginalSize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Originale</string>
   </property>
   <property name="toolTip">
    <string>taille originale</string>
   </property>
  </action>
  <action name="actionConstrainedSize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Custom</string>
   </property>
   <property name="iconText">
    <string>contrainte</string>
   </property>
   <property name="toolTip">
    <string>taille imposée</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quitter</string>
   </property>
  </action>
  <action name="actionRenderImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Image</string>
   </property>
  </action>
  <action name="actionRenderPixmap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pixmap</string>
   </property>
  </action>
  <action name="actionRenderOpenGL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>OpenGL</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QcvMatWidget</class>
   <extends>QWidget</extends>
   <header>QcvMatWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>radioButtonOrigSize</tabstop>
  <tabstop>radioButtonCustomSize</tabstop>
  <tabstop>spinBoxWidth</tabstop>
  <tabstop>spinBoxHeight</tabstop>
  <tabstop>checkBoxFlip</tabstop>
  <tabstop>comboBoxImages</tabstop>
  <tabstop>comboBoxEdges</tabstop>
  <tabstop>spinBoxKernel</tabstop>
  <tabstop>doubleSpinBoxSigma</tabstop>
  <tabstop>spinBoxThreshold</tabstop>
  <tabstop>doubleSpinBoxKappa</tabstop>
  <tabstop>horizontalSliderKernel</tabstop>
  <tabstop>horizontalSliderSigma</tabstop>
  <tabstop>horizontalSliderThreshold</tabstop>
  <tabstop>horizontalSliderKappa</tabstop>
  <tabstop>tabWidgetFilters</tabstop>
  <tabstop>scrollAreaSource</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>radioButtonCustomSize</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionConstrainedSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConstrainedSize</sender>
   <signal>triggered(bool)</signal>
   <receiver>radioButtonCustomSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>108</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonOrigSize</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionOriginalSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOriginalSize</sender>
   <signal>triggered(bool)</signal>
   <receiver>radioButtonOrigSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>108</x>
     <y>425</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFlip</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionFlip</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>386</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFlip</sender>
   <signal>triggered(bool)</signal>
   <receiver>checkBoxFlip</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>386</x>
     <y>425</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxKernel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSliderKernel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>630</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>573</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderKernel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBoxKernel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxThreshold</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSliderThreshold</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>467</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderThreshold</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBoxThreshold</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>470</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
